lib/elixir/lib/module/types/descr.ex

```diff
@@ -464,6 +464,26 @@ defmodule Module.Types.Descr do
 
   defp iterator_difference_static(:none, map), do: map
 
+  defp empty_difference_static?(left, :term), do: not Map.has_key?(left, :optional)
```

Comment:
@josevalim sorry asking for my understanding, couldn't figure this one out.

What would be an example case that justifies checking for `optional`?

Currently the following seems to pass all the tests, so it's probably not covered:
```diff
-  defp empty_difference_static?(left, :term), do: not Map.has_key?(left, :optional)
+  defp empty_difference_static?(left, :term), do: true
```